FreeRTOS/Demo/CORTEX_MPU_M3_MPS2_QEMU_GCC/init/startup.c

```diff
@@ -0,0 +1,232 @@
+    volatile uint32_t r1;
+    volatile uint32_t r2;
+    volatile uint32_t r3;
+    volatile uint32_t r12;
+    volatile uint32_t lr;  /* Link register. */
+    volatile uint32_t pc;  /* Program counter. */
+    volatile uint32_t psr; /* Program status register. */
+
+    r0 = pulFaultStackAddress[ 0 ];
+    r1 = pulFaultStackAddress[ 1 ];
+    r2 = pulFaultStackAddress[ 2 ];
+    r3 = pulFaultStackAddress[ 3 ];
+
+    r12 = pulFaultStackAddress[ 4 ];
+    lr = pulFaultStackAddress[ 5 ];
+    pc = pulFaultStackAddress[ 6 ];
+    psr = pulFaultStackAddress[ 7 ];
+
+    /* When the following line is hit, the variables contain the register values. */
+    for( ; ; )
+    {
+    }
+
+    /* remove the warning: variable <x> is set but not used */
+    ( void ) r0;
+    ( void ) r1;
+    ( void ) r2;
+    ( void ) r3;
+    ( void ) r12;
+    ( void ) lr;
+    ( void ) pc;
+    ( void ) psr;
+}
+
+static void Default_Handler( void ) __attribute__( ( naked ) );
+void Default_Handler( void )
+{
+    __asm volatile
+    (
+        "Default_Handler: \n"
+        "    ldr r3, NVIC_INT_CTRL_CONST  \n"
+        "    ldr r2, [r3, #0]\n"
+        "    uxtb r2, r2\n"
+        "Infinite_Loop:\n"
+        "    b  Infinite_Loop\n"
+        ".size  Default_Handler, .-Default_Handler\n"
+        ".align 4\n"
+        "NVIC_INT_CTRL_CONST: .word 0xe000ed04\n"
+    );
+}
+static void HardFault_Handler( void ) __attribute__( ( naked ) );
+void HardFault_Handler( void )
+{
+    __asm volatile
+    (
+        " tst lr, #4                                                \n"
+        " ite eq                                                    \n"
+        " mrseq r0, msp                                             \n"
+        " mrsne r0, psp                                             \n"
+        " ldr r1, [r0, #24]                                         \n"
+        " ldr r2, handler2_address_const                            \n"
+        " bx r2                                                     \n"
+        " handler2_address_const: .word prvGetRegistersFromStack    \n"
+    );
+}
+
+static void MemMang_Handler( void ) __attribute__( ( naked ) );
+void MemMang_Handler( void )
+{
+    __asm volatile
+    (
+        " tst lr, #4                                                         \n"
+        " ite eq                                                             \n"
+        " mrseq r0, msp                                                      \n"
+        " mrsne r0, psp                                                      \n"
+        " ldr r1, handler_address_const                                      \n"
+        " bx r1                                                              \n"
+        " handler_address_const: .word vHandleMemoryFault                    \n"
+    );
+}
+
+void BusFault_Handler( void )
+{
+    for( ; ; )
+    {
+    }
+}
+
+void UsageFault_Handler( void )
+{
+    for( ; ; )
+    {
+    }
+}
+
+void Debug_Handler( void )
+{
+    for( ; ; )
```

Comment:
Same as above.

Reply:
This is already an assert , we cannot print in this function